src/inputs/select/select.jsx

```diff
@@ -0,0 +1,831 @@
+        },
+    };
+});
+
+function Select(props) {
+    const {
+        className,
+        clearable: isClearable,
+        creatable: isCreatable,
+        disable: isDisabled,
+        dropdownMenuContainerStyle,
+        dropdownMenuStyle,
+        dropdownMenuMaxHeight,
+        dropdownMenuMinHeight,
+        fluid: isFluid,
+        id,
+        label,
+        matchProp,
+        multiple,
+        onChange: onChangeProp,
+        optionComponent,
+        options,
+        placeholder,
+        required,
+        searchable: isSearchable,
+        tabIndex,
+        underline: isUnderlined,
+        value,
+        valueComponent,
+    } = props;
+
+    const classes = useStyles(props);
+
+    const dropdownMenuRef = useRef();
+    const selectRef = useRef();
+
+    const onChange = (selectedOption) => {
+        console.log('onChangeProp', onChangeProp);
+        if (isFunction(onChangeProp)) {
+            console.log('selectedOption', selectedOption);
+            onChangeProp(selectedOption);
+        }
+    };
+
+    const menuRenderer = (params) => {
+        const items = map(params.options, (o, i) => {
+            if (optionComponent) {
+                const OptionComponent = optionComponent;
+
+                return (
+                    <OptionComponent
+                        isFocused={params.isFocused}
+                        key={`select-option-key-${i}`}
+                        onFocus={() => params.onFocus(o)}
+                        onSelect={() => params.selectValue(o)}
+                        option={o}
+                    />
+                );
+            }
+
+            return (
+                <div
+                    className="Select-option"
+                    key={`select-option-key-${i}`}
+                    onClick={() => params.selectValue(o)}
+                    onFocus={() => {}}
+                    onKeyDown={() => {}}
+                    onMouseOver={() => params.focusOption(o)}
+                    role="option"
+                    aria-selected="true"
+                    tabIndex={0}
+                >
+                    {o.label}
+                </div>
+            );
+        });
+
+        return (
+            <ScrollBar
+                autoHeight
+                autoHeightMax={dropdownMenuMaxHeight || 180}
+                autoHeightMin={dropdownMenuMinHeight}
+                autoHide
+                className="select-menu-scrollbar"
+                ref={dropdownMenuRef}
+            >
+                {items}
+            </ScrollBar>
+        );
+    };
+
+    const rootClasses = ClassNames(
+        UI_CLASS_NAME,
+        BEM_SELECT,
+        classes.root,
+        className,
+        {
+            [classes.isFluid]: isFluid,
+            [classes.isUnderlined]: isUnderlined,
+        },
+    );
+    const showRequiredIndicator = required && isEmpty(value);
+
+    const ReactSelectComponent = isCreatable ? ReactSelect.Creatable : CustomReactSelect;
```

Comment:
I'd rather see a blank line between 734 and 735, and think there's less call for one between 735 and 737, but to each his own.  _Not a Blocker_

Reply:
Fixed.

Reply:
Thanks for indulging my whitespace preferences! 😁 
Resolving.